cider/exec/processor/DefaultBatchProcessor.h

```diff
@@ -45,22 +49,26 @@ class DefaultBatchProcessor : public BatchProcessor,
   void feedHashBuildTable(const std::shared_ptr<JoinHashTable>& hashTable) override;
 
  protected:
-  DefaultBatchProcessor(const plan::SubstraitPlanPtr& plan,
-                        const BatchProcessorContextPtr& context);
-
-  const plan::SubstraitPlanPtr plan_;
+  plan::SubstraitPlanPtr plan_;
 
-  const BatchProcessorContextPtr context_;
+  BatchProcessorContextPtr context_;
 
   BatchProcessorState state_{BatchProcessorState::kRunning};
 
-  std::shared_ptr<CiderBatch> inputBatch_;
+  const struct ArrowArray* input_arrow_array_{nullptr};
```

Comment:
we need to consider how to release array and schema, may be we can use a wrap class to wrap the arrowarray and arrowschema like [duckdb] does (https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/common/arrow/arrow_wrapper.hpp)

Reply:
for input it is cider's responsibility to release array/schema, for output after the array/schema transferred to velox, the ownership also transferred to velox